src/bindings/jna/libelektra/src/test/java/org/libelektra/KeySetTest.java

```diff
@@ -814,4 +809,500 @@ public void test_keySetTailSetShouldPass() {
     assertTrue(keySet.contains(key5));
     assertTrue(tailSet.contains(key6));
   }
+
+  @Test
+  public void test_keySetLower_shouldPass() {
+    var keySet = KeySet.create(5, key2, key3, key4, key5);
+
+    assertNull(keySet.lower(key));
+    assertNull(keySet.lower(key2));
+    assertEquals(key2, keySet.lower(key3));
+    assertEquals(key3, keySet.lower(key4));
+    assertEquals(key4, keySet.lower(key5));
+    assertNull(keySet.lower(key6));
```

Comment:
This assertion is wrong. **All** of the elements in `keySet` are "strictly less than `key6`", so `keySet.lower(key6)` should return the last element of `keySet`.

Also there are no tests for the case, where the element is within the range of the `keySet` (i.e. between `keySet.first()` and `keySet.last()`) but not actually an element of `keySet`. An example would be:

```java
assertEquals(key2, keySet.lower("/key_test/2/key_name/1"));
```